### ISSUE TITLE
Map legend not showing 

### ISSUE THREAD

**Issue by janagombitova** [akvo/akvo-lumen#1579]
# Problem or idea
*What is the cause?*

Steps to reproduce: 
1. create a new map 
2. base first layer on shapes 
3. add another geopoint layer with a lot of data 
4. add what it should style on 
5. legend does not show properly in the UI

<img width="1259" alt="screen shot 2018-07-24 at 16 12 46" src="https://user-images.githubusercontent.com/12456965/43144321-208cd2ea-8f5d-11e8-9e1b-2aa538ea9349.png">

https://demo.akvolumen.org/s/xQ0uK4KKIyE

# Solution or next step
How could we solve it?

**Comment by tangrammer** [comment]
@janagombitova  I can't reproduce the issue 
trying similar steps here => https://demo.akvolumen.org/visualisation/5b8646b2-d267-4dab-957b-d164ffd44836

**Comment by tangrammer** [comment]
BTW: this url https://demo.akvolumen.org/s/xQ0uK4KKIyE seems not valid anymore

**Comment by janagombitova** [comment]
@tangrammer I tried again and what seems to do the trick is that if you have a gepshape layer and you select no Styling indicator and you also have a geopoint layer where you do select the styling indicator (what shows in the legend) then the legend does not show properly. 

If you however select also a styling indicator for the shape layer, then the legend shows. 

I tried to capture my steps here.
![maps legend](https://user-images.githubusercontent.com/12456965/44789065-f1cfad00-ab9b-11e8-9806-54e017107836.gif)
